data-prepper-plugins/saas-source-plugins/jira-source/src/main/java/org/opensearch/dataprepper/plugins/source/jira/rest/auth/JiraOauthConfig.java

```diff
@@ -147,9 +147,12 @@ public void renewCredentials() {
                         statusCode, ex.getMessage());
                 if (statusCode == HttpStatus.FORBIDDEN || statusCode == HttpStatus.UNAUTHORIZED) {
                     log.info("Trying to refresh the secrets");
-                    // Try refreshing the secrets and see if that helps
-                    // Refreshing one of the secret refreshes the entire store so we are good to trigger refresh on just one
-                    jiraSourceConfig.getAuthenticationConfig().getOauth2Config().getAccessToken().refresh();
+                    // Refreshing the secrets. It should help if someone already renewed the tokens.
+                    // Refreshing one of the secret refreshes the entire store so triggering refresh on just one
+                    oauth2Config.getAccessToken().refresh();
+                    this.accessToken = (String) oauth2Config.getAccessToken().getValue();
+                    this.refreshToken = (String) oauth2Config.getRefreshToken().getValue();
+                    this.expireTime = Instant.ofEpochMilli(System.currentTimeMillis() + (expiresInSeconds * 100L));
```

Comment:
1000L?
```suggestion
                    this.expireTime = Instant.ofEpochMilli(System.currentTimeMillis() + (expiresInSeconds * 1000L));
```


Or how about just:
```java
this.expireTime = Instant.now().plusSeconds(expiresInSeconds);
```

Reply:
This is definitely much more readable. I changed the code to follow your suggestion 👍